group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/TopicIds.java

```diff
@@ -164,20 +308,20 @@ public boolean equals(Object o) {
         TopicIds uuids = (TopicIds) o;
 
         if (!Objects.equals(topicNames, uuids.topicNames)) return false;
-        return Objects.equals(image, uuids.image);
+        return Objects.equals(resolver, uuids.resolver);
     }
 
     @Override
     public int hashCode() {
         int result = topicNames.hashCode();
-        result = 31 * result + image.hashCode();
+        result = 31 * result + resolver.hashCode();
         return result;
     }
 
     @Override
     public String toString() {
         return "TopicIds(topicNames=" + topicNames +
-            ", image=" + image +
+            ", resolver=" + resolver +
```

Comment:
nit: If we toString it here, should we add toString methods to the resolvers too?

Reply:
I'll add a toString implementation that returns `<class name>(image=<image>)`.